tenets/codelingo/go/string-format/codelingo.yaml

```diff
@@ -0,0 +1,30 @@
+functions:
+  - name: countVerbs
+    type: asserter
+    body: |
+      function (cc, str) {
+        var count = 1;
+        if (str[0] == '%') count++;
+        for (var i = 1; i < str.length; i++) {
+          if (str[i] == '%' && str[i-1] != '\\') count++;
+        }
+        return count != cc;
+      }
+tenets:
+  - name: find-funcs
+    doc: Find formatted strings missing variable(s).
+    flows:
+      codelingo/review:
+        comment: This formatted string is missing variable(s)
+    query: |
+      import codelingo/ast/go
+      @review.comment
+      go.call_expr(depth = any):
+        go.selector_expr
```

Comment:
Is `go.selector_expr` necessary? It seems like it's here to express the assumption that calls to functions in external packages (like `fmt` or `errors`) will require arguments for each format verb, whereas calls in the same package won't. 

This is decent and maybe you should keep it, but at least add comment explaining the following:
This will get some false negatives - here's an example from the standard library where a call to a local function (`Sprintf`) should have an argument for each verb https://golang.org/src/fmt/print.go?s=5747:5797#L201. This example brings up another point, which is that you should account for variadic args https://gobyexample.com/variadic-functions. Also I think there could be false positives, for example, you could pass an SQL query with format arguments into a function and not expect the verbs to be resolved.

Another possible heuristic would be to only have this apply to functions ending in "f" like "Errorf" or "Sprintf". This is obviously not perfect, but it can be improved when we add more NLP.